src/grisp_connect_api.erl

```diff
@@ -1,152 +1,76 @@
-% In case it was a response, returns the parsed ID and content to be handled by
-% the caller.
--spec handle_msg(JSON :: binary()) ->
-    {send_response, Response :: binary()} |
-    {handle_response, ID :: binary(), {ok, Result :: map()} | {error, atom()}}.
-handle_msg(JSON) ->
-    JSON_RPC = grisp_connect_jsonrpc:decode(JSON),
-    handle_jsonrpc(JSON_RPC).
 
 %--- Internal Funcitons --------------------------------------------------------
 
-format_error({internal_error, parse_error, ID}) ->
-    {error, -32700, <<"Parse error">>, undefined, ID};
-format_error({internal_error, invalid_request, ID}) ->
-    {error, -32600, <<"Invalid request">>, undefined, ID};
-format_error({internal_error, method_not_found, ID}) ->
-    {error, -32601, <<"Method not found">>, undefined, ID};
-format_error({internal_error, invalid_params, ID}) ->
-    {error, -32602, <<"Invalid params">>, undefined, ID};
-format_error({internal_error, Reason, ID}) ->
-    {error, -32603, <<"Internal error">>, Reason, ID}.
-
-%FIXME: Batch are not supported yet. When receiving a batch of messages, as per
-%       the JSON-RPC standard, all the responses should goes in a single batch
-%       of responses.
-handle_jsonrpc(Messages) when is_list(Messages) ->
-    handle_rpc_messages(Messages, []);
-handle_jsonrpc(Message) ->
-    handle_rpc_messages([Message], []).
-
-handle_rpc_messages([], Replies) -> lists:reverse(Replies);
-handle_rpc_messages([{request, M, Params, ID} | Batch], Replies)
-  when M == ?method_post;
-       M == ?method_get ->
-    handle_rpc_messages(Batch, [handle_request(M, Params, ID) | Replies]);
-handle_rpc_messages([{result, _, _} = Res| Batch], Replies) ->
-    handle_rpc_messages(Batch, [handle_response(Res)| Replies]);
-handle_rpc_messages([{error, _Code, _Msg, _Data, _ID} = E | Batch], Replies) ->
-    ?LOG_INFO("Received JsonRPC error: ~p",[E]),
-    handle_rpc_messages(Batch, [handle_response(E)| Replies]);
-handle_rpc_messages([{decoding_error, _, _, _, _} = E | Batch], Replies) ->
-    ?LOG_ERROR("JsonRPC decoding error: ~p",[E]),
-    handle_rpc_messages(Batch, Replies).
-
-handle_request(?method_get, #{type := <<"system_info">>} = _Params, ID) ->
+handle_request([?method_get], #{type := <<"system_info">>} = _Params, ID) ->
     Info = grisp_connect_updater:system_info(),
-    {send_response, grisp_connect_jsonrpc:encode({result, Info, ID})};
-handle_request(?method_post, #{type := <<"start_update">>} = Params, ID) ->
+    {reply, Info, ID};
+handle_request([?method_post], #{type := <<"start_update">>} = Params, ID) ->
     try
         URL = maps:get(url, Params),
-        Reply = case grisp_connect_updater:start_update(URL) of
+        case grisp_connect_updater:start_update(URL) of
             {error, grisp_updater_unavailable} ->
-                {error, -10, grisp_updater_unavailable, undefined, ID};
+                {error, grisp_updater_unavailable, undefined, undefined, ID};
```

Comment:
All the error tuples are not handled as return values is `grisp_connect_client:handle_connection_message/2`.